components/LevitatingLink.js

```diff
@@ -1,24 +1,33 @@
-export default function LevitatingLink ({ href, children }) {
+import { Link } from 'react-scroll'
+
+export default function LevitatingLink ({ href, withScrollTarget, children }) {
```

Comment:
I think, this is not the right place to implement the smooth scrolling logic, because this will limit the smooth scrolling ability to the LevitatingLink.

Generally, when the user clicks on any link, whether LevitatingLink or not, the website should detect if it is an internal link, then scroll to it.

Instead of using the provided components, you can use its [Scroll method utility function](https://github.com/fisshy/react-scroll#scroll-methods), `Scroll.animateScroll.scrollTo(element, { ...options })`, by adding an `onClick` listener in our `MainLayout`.

The onclick listener should do this:

- Check if the user is clicking a clink (by checking if `e.target` or its ancestors is an `A` tag).
- If the user is clicking a link, check if it’s an internal link.
- If it’s an internal link, make sure that user is not pressing Alt, Ctrl, Shift, or Meta keys when clicking.
- Take over browser’s action by calling `e.preventDefault()`.
- Then make the browser scroll smoothly to it.

Another possible alternative is to use a lightweight [Smooth Scroll](https://cferdinandi.github.io/smooth-scroll/index.html) library. It does not require jQuery. If you go this route, make sure to remove `react-scroll`.